CHANGELOG.md

```diff
@@ -4,6 +4,494 @@ All notable changes to Router will be documented in this file.
+
+### Expose query plan in extensions for GraphQL response (experimental) ([PR #1470](https://github.com/apollographql/router/pull/1470))
+
+Expose query plan in extensions for GraphQL response. Only experimental for now, no documentation available.
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1470
+
+### Add support of global rate limit and timeout. [PR #1347](https://github.com/apollographql/router/pull/1347)
+
+Additions to the traffic shaping plugin:
+- **Global rate limit** - If you want to rate limit requests to subgraphs or to the router itself.
+- **Timeout**: - Set a timeout to subgraphs and router requests.
+
+```yaml
+traffic_shaping:
+  router: # Rules applied to requests from clients to the router
+    global_rate_limit: # Accept a maximum of 10 requests per 5 secs. Excess requests must be rejected.
+      capacity: 10
+      interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+    timeout: 50s # If a request to the router takes more than 50secs then cancel the request (30 sec by default)
+  subgraphs: # Rules applied to requests from the router to individual subgraphs
+    products:
+      global_rate_limit: # Accept a maximum of 10 requests per 5 secs from the router. Excess requests must be rejected.
+        capacity: 10
+        interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+      timeout: 50s # If a request to the subgraph 'products' takes more than 50secs then cancel the request (30 sec by default)
+```
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1347
+
+### Explicit `shutdown` for `RouterHttpServer` handle ([PR #1487](https://github.com/apollographql/router/pull/1487))
+
+If you explicitly create a `RouterHttpServer` handle,
+dropping it while the server is running instructs the server shut down gracefuly.
+However with the handle dropped, there is no way to wait for shutdown to end
+or check that it went without error.
+Instead, the new `shutdown` async method can be called explicitly
+to obtain a `Result`:
+
+```diff
+ use RouterHttpServer;
+ let server = RouterHttpServer::builder().schema("schema").start();
+ // …
+-drop(server);
++server.shutdown().await.unwrap(); 
+```
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Added `apollo_router::TestHarness` ([PR #1487](https://github.com/apollographql/router/pull/1487))
+
+This is a builder for the part of an Apollo Router that handles GraphQL requests,
+as a `tower::Service`.
+This allows tests, benchmarks, etc
+to manipulate request and response objects in memory without going over the network.
+See the API documentation for an example. (It can be built with `cargo doc --open`.)
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Remove telemetry configuration hot reloading ([PR #1501](https://github.com/apollographql/router/pull/1501))
+
```

Comment:
@Geal This header is not matching the text (removing hot-reloading still exists above in breaking changes).  Maybe we want this:

```suggestion
### Introduce `map_deferred_response` method for deferred responses ([PR #1501](https://github.com/apollographql/router/pull/1501))

```

Reply:
right, good catch